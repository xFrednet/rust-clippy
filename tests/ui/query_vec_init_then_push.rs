#![allow(clippy::all)]
#![deny(clippy::graph_query_linter)]

fn main() {
    let mut def_err: Vec<u32> = Default::default();
    def_err.push(0);

    let mut new_err = Vec::<u32>::new();
    new_err.push(1);

    let mut cap_err = Vec::with_capacity(2);
    cap_err.push(0);
    cap_err.push(1);
    cap_err.push(2);
    if true {
        // don't include this one
        cap_err.push(3);
    }

    let mut cap_ok = Vec::with_capacity(10);
    cap_ok.push(0);

    new_err = Vec::new();
    new_err.push(0);

    let mut vec = Vec::new();
    // control flow at block final expression
    if true {
        // no lint
        vec.push(1);
    }
}
