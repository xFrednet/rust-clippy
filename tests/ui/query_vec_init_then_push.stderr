error: calls to `push` immediately after creation
  --> $DIR/query_vec_init_then_push.rs:8:5
   |
LL | /     let mut new_err = Vec::<u32>::new();
LL | |     new_err.push(1);
   | |___________________^ help: consider using the `vec![]` macro: `let mut new_err = vec![..];`
   |
note: the lint level is defined here
  --> $DIR/query_vec_init_then_push.rs:2:9
   |
LL | #![deny(clippy::graph_query_linter)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: calls to `push` immediately after creation
  --> $DIR/query_vec_init_then_push.rs:5:5
   |
LL | /     let mut def_err: Vec<u32> = Default::default();
LL | |     def_err.push(0);
   | |___________________^ help: consider using the `vec![]` macro: `let mut def_err = vec![..];`

error: calls to `push` immediately after creation
  --> $DIR/query_vec_init_then_push.rs:11:5
   |
LL | /     let mut cap_err = Vec::with_capacity(2);
LL | |     cap_err.push(0);
   | |___________________^ help: consider using the `vec![]` macro: `let mut cap_err = vec![..];`

error: calls to `push` immediately after creation
  --> $DIR/query_vec_init_then_push.rs:20:5
   |
LL | /     let mut cap_ok = Vec::with_capacity(10);
LL | |     cap_ok.push(0);
   | |__________________^ help: consider using the `vec![]` macro: `let mut cap_ok = vec![..];`

error: calls to `push` immediately after creation
  --> $DIR/query_vec_init_then_push.rs:23:5
   |
LL | /     new_err = Vec::new();
LL | |     new_err.push(0);
   | |___________________^ help: consider using the `vec![]` macro: `new_err = vec![..];`

error: aborting due to 5 previous errors

